.github/BRANCH_AND_RELEASE_PROCESS.md

```diff
@@ -0,0 +1,88 @@
+
+   This strategy emphasizes simplicity and continuous integration, making it well-suited for fast-paced development environments.
+
+3. **Trunk-Based Development**:
+
+   Trunk-Based Development focuses on keeping a single, stable branch (the "trunk") where all developers commit their code. Key practices include:
+
+   - Small, frequent commits directly to the `main` branch.
+   - Short-lived feature branches that are quickly merged back into `main`.
+   - Emphasis on automated testing and continuous integration to ensure code stability.
+   This strategy aims to minimize merge conflicts and maintain a high level of code quality, promoting rapid feedback and collaboration.
+
+Each of these strategies has its own strengths and is chosen based on the specific needs and workflow of the project.
+
+## Branch Process for This Project
+
+This project's branch process sits between **GitHub Flow** and **Git Flow** by taking the best of both worlds. This projects branching strategy looks like:
+
+Aspects used from **GitHub Flow**:
+
+- A single `main` branch always containing deployable code.
+- Branches for each feature or bug fix that branch off from `main` and merge back into `main` upon completion.
+- Continuous deployment from the `main` branch, allowing for fast iterations and rapid delivery of new features.
+
+Aspects used from **Git Flow**:
+
+- `release-v[0-9]+/*`: Branches off from `main` when preparing a new release.
+- `hotfix/*`: Branches off from `main` (or a release branch) for critical fixes that need to be deployed immediately.
+
+### Why Pick This Strategy?
+
+This is done in order to foster:
+
+- maximum collaboration with external contributors
+  - since we are on GitHub, it's the standard workflow by default. (its why `develop`, or `alpha`/`beta`/etc branches aren't created by default)
+  - it's intuitively obvious where contributions (ie PRs) need to merge with zero background on the project
+  - this puts all bespoke, project, and repo management on the project maintainers
+- forces the project maintainers to embrace CI/CD
+  - `main` must work at all times; therefore, main can be deployed or released at all times
+- things don't always go according to plan
+  - having the branching strategy for releases **Git Flow** helps support of concurrent versions
+  - provides flexibilty to create release trains
+
+## Release Process
+
+The release process for this project is designed to balance the rapid iteration capabilities of **GitHub Flow** with the structured release management of **Git Flow**. Releases are typically created off `main` since we strive to keep backwards compatibility and prevent breaking any interfaces. This implies that releases are basically a single train pushing features out. In terms new feature release health, you should consider `main` branch unstable. Consumers of this SDK should **ONLY** ever consume a tagged release on the repo release page.
```

Comment:
```suggestion
The release process for this project is designed to balance the rapid iteration capabilities of **GitHub Flow** with the structured release management of **Git Flow**. Releases are typically created off `main` since we strive to keep backwards compatibility and prevent breaking any interfaces. This implies that releases are basically a single train pushing features out. In terms of new feature release health, you should consider the `main` branch unstable. Consumers of this SDK should **ONLY** ever consume a tagged release on the repo release page.
```